src/app/pages/BuyCrypto/BuyCrypto.tsx

```diff
@@ -1,16 +1,113 @@
 import { ExchangeDataInterface } from 'lib/exolix-api';
 import { T, t } from 'lib/i18n/react';
 import { useAccount, useNetwork, useStorage } from 'lib/temple/front';
-import { Redirect } from 'lib/woozie';
+import { Redirect, Link } from 'lib/woozie';
+
+import { BuyCryptoSelectors } from './BuyCypto.selectors';
+import { Cryptos } from './Cryptos';
+import { Debits } from './Debits';
+import { useTabSlug } from './hooks/use-tab-slug';
+
+interface BuyCryptoProps {
+  crypto?: boolean;
+}
+
+const BuyCrypto: FC<BuyCryptoProps> = ({ crypto = false }) => {
+  const { fullPage } = useAppEnv();
+  const tabSlug = useTabSlug();
+
+  const tabs = useMemo<
+    {
+      slug: string;
+      title: string;
+      Component: FC;
+      testID: string;
+    }[]
+  >(() => {
+    return [
+      {
+        slug: 'crypto',
+        title: t('topUpCrypto'),
+        Component: Cryptos,
+        testID: BuyCryptoSelectors.Crypto
+      },
+      {
+        slug: 'debit',
+        title: t('topUpDebit'),
+        Component: Debits,
+        testID: BuyCryptoSelectors.Debit
+      }
+    ];
+  }, []);
+
+  const { slug, Component } = useMemo(() => {
+    const tab = tabSlug ? tabs.find(currentTab => currentTab.slug === tabSlug) : null;
+    return tab ?? tabs[0];
+  }, [tabSlug, tabs]);
+
+  if (crypto) {
+    return <BuyCryptoOld />;
+  }
+  return (
+    <PageLayout
+      pageTitle={
+        <>
+          <T id="topUpBuy" />
+        </>
+      }
+    >
+      <div className="text-center my-3">
+        <T id="topUpDescription" />
+      </div>
+      <div className={classNames('-mx-4', 'shadow-top-light', fullPage && 'rounded-t-md')}>
+        <div className={classNames('w-full max-w-sm mx-auto px-10', 'flex items-center justify-center')}>
+          {tabs.map(currentTab => {
+            const active = slug === currentTab.slug;
+
+            return (
+              <Link
+                key={currentTab.slug}
+                to={lctn => ({ ...lctn, search: `?tab=${currentTab.slug}` })}
+                replace
+                className={classNames(
+                  'flex1 w-full',
+                  'text-center cursor-pointer mb-1 pb-1 pt-2',
+                  'text-gray-500 text-xs font-medium',
+                  'border-b-2',
+                  active ? 'border-primary-orange' : 'border-transparent',
+                  active ? 'text-primary-orange' : 'hover:text-primary-orange',
+                  'transition ease-in-out duration-300',
+                  'truncate'
+                )}
+                testID={currentTab.testID}
+              >
+                {currentTab.title}
+              </Link>
+            );
+          })}
+        </div>
 
-const BuyCrypto: FC = () => (
+        <div className={'mx-4 mb-4 mt-6'}>
+          <SuspenseContainer whileMessage="displaying tab">{Component && <Component />}</SuspenseContainer>
+        </div>
+      </div>
+    </PageLayout>
+  );
+};
+
+const BuyCryptoOld: FC = () => (
```

Comment:
rename to `BuyWithSomething`, as the current name does not explain what this component does.